### DISCUSSION TITLE
Django IntegerField Enumeration does not correctly convert the __empty__ value in a ChoiceField

### DISCUSSION THREAD

**Discussion by gabn88** [encode/django-rest-framework#9113] [Question & Answer]
## Checklist

- [X] Raised initially as discussion #...
- [X] This cannot be dealt with as a third party library. (We prefer new functionality to be [in the form of third party libraries](https://www.django-rest-framework.org/community/third-party-packages/#about-third-party-packages) where possible.)
- [X] I have reduced the issue to the simplest possible case.

Giving the following model and a default Serializer configuration, this raise a TypeError if you select 'unknown' in the Browsable API.

The error: `'<' not supported between instances of 'NoneType' and 'int'` seems to be coming from the `MinValueValidator`.

```
class MyChoices(models.IntegerChoices):
    ONE = 1, 'One'
    TWO = 2, 'Two'
    __empty__ =  'unknown'

class MyModel(models.Model):
    something = models.IntegerField(choices=MyChoices.choices, null=True, blank=True)
```

The problems seems to be coming from `fields.py`:
```
    def run_validation(self, data=empty):
        """
        Validate a simple representation and return the internal value.

        The provided data may be `empty` if no representation was included
        in the input.

        May raise `SkipField` if the field should not be included in the
        validated data.
        """
        (is_empty_value, data) = self.validate_empty_values(data) -> data is 'None' here, so the is_empty_value converts to False
        if is_empty_value:
            return data
        value = self.to_internal_value(data)
        self.run_validators(value)
        return value
```

**Comment by Ehsan200** [comment]
Hello there!
I've just tested this in browsable API and there is no problem occurred for me with these codes:

```python

# serializers.py
class MyModelSerializer(ModelSerializer):
    class Meta:
        model = MyModel
        fields = '__all__'

# views.py
class MyModelViewSet(ModelViewSet):
    serializer_class = MyModelSerializer
    queryset = MyModel.objects.all()

# urls.py
router = routers.SimpleRouter()

router.register(r'my-model', MyModelViewSet)

urlpatterns = [
    *router.urls,
]

```
<img width="964" alt="Screenshot 2023-09-19 at 20 32 56" src="https://github.com/encode/django-rest-framework/assets/59165228/8f17c3d4-fe33-4636-898f-b3efb958a767">

could you please provide us with the version of Django and DRF used?
my versions:
-Django              4.1.6
-djangorestframework 3.14.0
